src/AdventureGoldGovernance.sol

```diff
@@ -83,11 +87,24 @@ contract AdventureGoldGovernance is IERC6372, ERC20Permit, ERC20Votes {
         emit Withdrawal(msg.sender, amount);
     }
 
-    /// @notice Only transfers to/from this contract are allowed
+    function transfer(address to, uint256 value) public override(ERC20) returns (bool) {
+        revert NonTransferable();
+    }
+
+    function transferFrom(address from, address to, uint256 value) public override(ERC20) returns (bool) {
+        revert NonTransferable();
+    }
+
     /// @dev For Solidity lineraization, see https://medium.com/@kalexotsu/inheritance-inheritance-order-and-the-super-keyword-in-solidity-bbe49a2478b6
     /// @dev ERC20Votes is called since Solidity inheritance is linearized from right to left
     function _update(address from, address to, uint256 value) internal override(ERC20, ERC20Votes) {
-        if (from != address(this) || to != address(this)) revert OnlyTransfersToFromContract();
+        // Deny the transfer if:
+        // - The `from` address is not the zero address or this contract
+        // Or:
+        // - The `to` address is not the zero address or this contract
+        if (!((from == address(this) || from == address(0)) || (to == address(this) || to == address(0)))) {
```

Comment:
Should this be this instead ?:
`if (!((from == address(this) && to == address(0)) || (from == address(0) && to == address(this))))` ?

This ensures that transfers can only be `address(this) -> address(0)` and `address(0) -> address(this)`.

Reply:
Your current condition can allow `address(this) -> address(this)` and `address(0) -> address(0)` also.